app/src/ui/blank-slate/create-tutorial-repository-dialog.tsx

```diff
@@ -0,0 +1,279 @@
+
+    try {
+      const path = Path.resolve(getDefaultDir(), name)
+
+      if (await pathExists(path)) {
+        throw new Error(
+          `The path ${path} already exists. Please move it ` +
+            'out of the way, or remove it, and then try again.'
+        )
+      }
+
+      this.setProgress(`Creating repository on ${endpointName}`, 0)
+
+      const repo = await this.createAPIRepository(account, name)
+
+      this.setProgress('Initializing local repository', 0.2)
+
+      await ensureDir(path)
+      await git(['init'], path, 'tutorial:init')
+
+      await writeFile(Path.join(path, 'README.md'), InititalReadmeContents)
+
+      await git(['add', '--', 'README.md'], path, 'tutorial:add')
+      await git(
+        ['commit', '-m', 'Initial commit', '--', 'README.md'],
+        path,
+        'tutorial:commit'
+      )
+      await git(
+        ['remote', 'add', 'origin', repo.clone_url],
+        path,
+        'tutorial:add-remote'
+      )
+
+      await this.pushRepo(path, account, (title, value, description) => {
+        this.setProgress(title, 0.3 + value * 0.7, description)
+      })
+
+      this.setState({ progress: undefined })
+      this.props.onTutorialRepositoryCreated(path, account, repo)
+      this.props.onDismissed()
+    } catch (err) {
+      if (err instanceof GitError) {
+        this.props.onError(err)
+      } else {
+        this.props.onError(
+          new Error(
+            `Failed creating the tutorial repository.\n\n${err.message}`
+          )
+        )
+      }
+    } finally {
+      this.setState({ loading: false })
+    }
+  }
+
+  private setProgress(title: string, value: number, description?: string) {
+    this.setState({
+      progress: { kind: 'generic', title, value, description },
+    })
+  }
+
+  public onCancel = () => {
+    this.props.onDismissed()
+  }
+
+  private renderProgress() {
+    if (this.state.progress === undefined) {
+      return null
+    }
+
+    const { progress } = this.state
+    const description = progress.description ? (
+      <div className="description">{progress.description}</div>
+    ) : null
+
+    return (
+      <div className="progress-container">
+        <div>{progress.title}</div>
+        <progress value={progress.value} />
+        {description}
+      </div>
+    )
+  }
+
+  public render() {
+    const { account } = this.props
+
+    return (
+      <Dialog
+        id="create-tutorial-repository-dialog"
+        title="Start tutorial"
+        onDismissed={this.onCancel}
+        onSubmit={this.onSubmit}
+        dismissable={!this.state.loading}
+        loading={this.state.loading}
+      >
+        <DialogContent>
+          <div>
+            This will create a repository on your local machine, and push it to
+            your account <Ref>@{this.props.account.login}</Ref> on{' '}
+            <LinkButton uri={getHTMLURL(account.endpoint)}>
+              {friendlyEndpointName(account)}
+            </LinkButton>
+            .
```

Comment:
```suggestion
            . The repository will be private and only visible to you.
```

Reply:
I had forgotten to push this, sorry. I used the text that @ampinsk added in her comment: https://github.com/desktop/desktop/issues/8148#issuecomment-524076065